### PR TITLE
fix: avm/ptn/data/private-analytical-workspace - failing tests with random locations

### PR DESCRIPTION
**Pull request by jbinko**
## Description

Fixes #3307

## Pipeline Reference

| Pipeline |
| -------- |
[avm.ptn.data.private-analytical-workspace](https://github.com/jbinko/bicep-registry-modules/actions/workflows/avm.ptn.data.private-analytical-workspace.yml)

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [x] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: .github/workflows/avm.ptn.data.private-analytical-workspace.yml [hunk 1]
```diff
@@ -22,6 +22,7 @@ on:
         type: string
         description: "Default location overwrite (e.g., eastus)"
         required: false
+        default: 'northeurope'
```
Comment (AlexanderSehr) [inline .github/workflows/avm.ptn.data.private-analytical-workspace.yml:25]: Please don't hardcode the value in the workflow (which overwrites the location in all tests), but rather replace the default resource location parameter 
```bicep
@description('Optional. The location to deploy resources to.')
param resourceLocation string = deployment().location
```
with 
```bicep
// enforcing location due to quote restrictions
#disable-next-line no-hardcoded-location
var enforcedLocation = 'northeurope'
```
in the `main.test.bicep` files and update the references. This allows you/us to specify different locations for different tests and is more in line with other, similar cases like in this module

https://github.com/Azure/bicep-registry-modules/blob/a0c516bd1d7b34f4a09278df077a25f15a89a2c6/avm/res/net-app/net-app-account/tests/e2e/defaults/main.test.bicep#L14-L16

**Comment by AlexanderSehr** [comment]
Hey @jbinko,
the PR looks good from my side. There just seems to be a small issue with 2 post deployment tests that need fixing. Then we should be good to go. Thank you for all the effort 💪

**Comment by jbinko** [comment]
> Hey @jbinko, the PR looks good from my side. There just seems to be a small issue with 2 post deployment tests that need fixing. Then we should be good to go. Thank you for all the effort 💪

Thanks @AlexanderSehr - Yes, I was working on it meanwhile. :-) It was Azure backend complaining about some not fully removed resources from the previous run. Now it should be fine.

**Review by AlexanderSehr — APPROVED** [review]

**Comment by AlexanderSehr** [comment]
Great work. Thanks @jbinko. Off we go 🚀

**Pull request merged**